graphql_compiler/schema_transformation/rename_schema.py

```diff
@@ -388,8 +602,161 @@ def enter_field_definition(
+    ) -> None:
+        """Record the current type that the visitor is traversing."""
+        self.current_type = node.name.value
+
+    def leave_object_type_definition(
+        self,
+        node: ObjectTypeDefinitionNode,
+        key: Any,
+        parent: Any,
+        path: List[Any],
+        ancestors: List[Any],
+    ) -> None:
+        """Finish traversing the current type node."""
+        self.current_type = None
+
+    def enter_field_definition(
+        self,
+        node: FieldDefinitionNode,
+        key: Any,
+        parent: Any,
+        path: List[Any],
+        ancestors: List[Any],
+    ) -> None:
+        """Check that no type Y contains a field of type X, where X is suppressed."""
+        if self.current_type == self.query_type:
+            return IDLE
+        # At a field of a type that is not the query type
+        field_name = node.name.value
+        field_type = get_ast_with_non_null_and_list_stripped(node.type).name.value
+        if self.renamings.get(field_type, field_type):
+            return IDLE
+        # Reaching this point means this field is of a type to be suppressed.
+        if self.current_type is None:
+            raise AssertionError(
+                "Entered a field not in any ObjectTypeDefinition scope because "
+                "self.current_type is None"
+            )
+        if self.current_type == field_type:
+            # Then node corresponds to a field belonging to type T that is also of type T.
+            # Therefore, we don't need to explicitly suppress the field as well and this should not
+            # raise errors.
+            return IDLE
+        if self.current_type not in self.fields_to_suppress:
+            self.fields_to_suppress[self.current_type] = {}
+        self.fields_to_suppress[self.current_type][field_name] = field_type
+        return IDLE
+
+    def enter_union_type_definition(
+        self,
+        node: UnionTypeDefinitionNode,
+        key: Any,
+        parent: Any,
+        path: List[Any],
+        ancestors: List[Any],
+    ) -> None:
+        """Check that each union still has at least one non-suppressed member."""
+        union_name = node.name.value
+        # Check if all the union members are suppressed.
+        for union_member in node.types:
+            union_member_type = get_ast_with_non_null_and_list_stripped(union_member).name.value
+            if self.renamings.get(union_member_type, union_member_type):
+                # Then at least one member of the union is not suppressed, so there is no cascading
+                # suppression error concern.
+                return IDLE
+        if self.renamings.get(union_name, union_name) is None:
+            # If the union is also suppressed, then nothing needs to happen here
+            return IDLE
+        self.union_types_to_suppress.append(node)
+
+
+class SuppressionNotImplementedVisitor(Visitor):
+    def __init__(self, renamings: Mapping[str, Optional[str]]) -> None:
```

Comment:
Same comment as above with regard to the intended API of this object and its documentation.